File: greatest-common-divisor/greatest_common_divisor.py
def greatest_common_divisor(p, q):
    if q is 0:
        return p
    else:
        r = p % q
        return greatest_common_divisor(q, r)

if __name__ == "__main__":
    result = greatest_common_divisor(8, 4)
    print(result)

File: greatest-common-divisor/GreatestCommonDivisor.java
class GreatestCommonDivisor
{
    public static void main(String args[])
    {
        int result = gcd(8, 4);
        System.out.println(result);
    }

    public static int gcd(int p, int q)
    {
    	if (q == 0) return p;
    	int r = p % q;
    	return gcd(q, r);
    }
}